app/scripts/background.js

```diff
@@ -586,10 +614,11 @@ export function setupController(
           }
         });
       }
-
+      const id = `${remotePort.sender?.tab?.id}-${processName}`;
```

Comment:
1. this generation happens multiple times, make a function out of it
2. if `remotePort.sender?.tab?.id` or `processName` are undefined for some reason, throw exception (otherwise attackers might abuse such a scenario)